### PR TITLE
Refs #18100, #26207 -- Added tests for deferred model deletion signals.

### PR DESCRIPTION
**Pull request by charettes**
Thanks Tim for pointing out this was fixed by 7f51876f99851fdc3fef63ae.

### CODE REVIEW

File: tests/defer_regress/tests.py [hunk 1]
```diff
@@ -238,3 +239,41 @@ def test_defer_annotate_select_related(self):
                  .defer('request1', 'request2', 'request3', 'request4')),
             list
         )
+
+
+class DeferDeletionSignalsTests(TestCase):
+    senders = [Item, Proxy]
+
+    def setUp(self):
+        super(DeferDeletionSignalsTests, self).setUp()
```
Comment (timgraham) [inline tests/defer_regress/tests.py]: No need for this?

File: tests/defer_regress/tests.py [hunk 2]
```diff
@@ -238,3 +239,41 @@ def test_defer_annotate_select_related(self):
...
+        self.clear_received_senders()
```
Comment (timgraham) [inline tests/defer_regress/tests.py]: inline the assignments rather than creating a separate method?

Comment (charettes) [inline tests/defer_regress/tests.py]: I realized why it was a method, it should be called in the test. Will fix.

File: tests/defer_regress/tests.py [hunk 3]
```diff
@@ -238,3 +239,40 @@ def test_defer_annotate_select_related(self):
...
+    def post_delete_receiver(self, sender, **kwargs):
+        self.post_delete_senders.append(sender)
+
+    def test_delete_defered_model(self):
+        item = Item.objects.create(value=1)
+        Item.objects.only('value').get(pk=item.pk).delete()
+        self.assertEqual(self.pre_delete_senders, [Item])
+        self.assertEqual(self.post_delete_senders, [Item])
+
+        self.clear_received_senders()
```
Comment (timgraham) [inline tests/defer_regress/tests.py]: Any disadvantage to making it a separate test method instead? I guess the signal connecting might be better is `setUpClass` at that point.

Comment (charettes) [inline tests/defer_regress/tests.py]: Receivers can't be connected in `setUpClass()` as they need to store the received senders on the test instance itself. I split the test in two methods and used `setUpTestData()`

**Pull request merged**